File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(const char* s, const char* skip, int index) {
    int len = strlen(s);
    char* answer = (char*)malloc((len + 1) * sizeof(char));
    if (answer == NULL) return NULL;
    
    for (int i = 0; i < len; i++) {
        char c = s[i];
        int count = 0;
        while (count < index) {
            c++;
            
            if (c > 'z') {
                c = 'a';
            }
            
            if (strchr(skip, c) == NULL) {
                count++;
            }
        }
        answer[i] = c;
    }
    answer[len] = '\0';
    
    return answer;
}

File: 프로그래머스/lv1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ""
    for char in s:
        count = 0
        while count < index:
            char = chr(ord(char) + 1)
            
            if char > 'z':
                char = 'a'
            
            if char not in skip:
                count += 1
                
        answer += char
        
    return answer